### PR TITLE
Optimize IMap VALUES INSERTs

### PR DESCRIPTION
**Pull request by gierlachg**

### CODE REVIEW

**Comment by gierlachg** [comment]
run-lab-run

**Review by ivanthescientist — APPROVED** [review]

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/SqlPrimitiveTest.java [hunk 1]
```diff
@@ -300,6 +300,22 @@ public void when_insert() {
...
+                "INSERT INTO " + name + " (this, __key) VALUES ('1', 1)",
+                createMap(1, "1")
+        );
+        assertRowsAnyOrder(
+                "SELECT * FROM " + name,
+                singletonList(new Row(1, "1"))
+        );
+    }
+
+    @Test
```
Comment (viliam-durina) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/SqlPrimitiveTest.java:314]: This test added to this class fails:
```java
    @Test
    public void test() {
        String name = randomName();
        sqlService.execute(javaSerializableMapDdl(name, String.class, String.class));

        assertEquals(1, sqlService.execute("INSERT INTO " + name + " SELECT * FROM (VALUES ('1', '1')) AS t(a, b) WHERE a='0'").updateCount());
    }
```

**Review by viliam-durina — CHANGES_REQUESTED** [review]

**Review by viliam-durina — APPROVED** [review]

**Comment by gierlachg** [comment]
run-lab-run

**Comment by gierlachg** [comment]
run-lab-run

**Pull request merged**